pkg/images/integration/util.go

```diff
@@ -23,14 +24,14 @@ func GetMatchingImageIntegrations(registries []registryTypes.ImageRegistry,
 	sort.Slice(matchingIntegrations, func(i, j int) bool {
 		// Note: the Name of ImageRegistry does not reflect the registry hostname used within the integration but a
 		// name chosen by the creator. Additionally, we have to trim the HTTP prefixes (http:// & https://).
-		if matchingIntegrations[i].Config() == nil {
+		if matchingIntegrations[i].Config(ctx) == nil {
 			return true
 		}
-		if matchingIntegrations[j].Config() == nil {
+		if matchingIntegrations[j].Config(ctx) == nil {
 			return false
 		}
-		return urlfmt.TrimHTTPPrefixes(matchingIntegrations[i].Config().RegistryHostname) <
-			urlfmt.TrimHTTPPrefixes(matchingIntegrations[j].Config().RegistryHostname)
+		return urlfmt.TrimHTTPPrefixes(matchingIntegrations[i].Config(ctx).RegistryHostname) <
+			urlfmt.TrimHTTPPrefixes(matchingIntegrations[j].Config(ctx).RegistryHostname)
```

Comment:
`Config(ctx)` is essentially already cached and just returns a struct, unless it got stale. Now here we could ignore the staleness and add another cache layer on top since the hostname is always fresh. But tbh I don't think it's worth it to bother with that, and it could potentially lead to a nasty bug if we add a cache and invalidate it incorrectly.